### DISCUSSION TITLE
x:DataType RelativeSource binding warning

### DISCUSSION THREAD

**Discussion by IeuanWalker** [dotnet/maui#25957] [Q&A]
Since .NET 9 update we are now getting these warnings -
```
Binding could be compiled to improve runtime performance if x:DataType is specified. See https://learn.microsoft.com/dotnet/maui/fundamentals/data-binding/compiled-bindings for more information.
```

I understand how to fix it for normal bindings but how to i fix it for `RelativeSource` bindings?
![image](https://github.com/user-attachments/assets/850ad5b6-30c5-4b27-b91c-e6680ee873ac)

**Comment by IeuanWalker** [comment]
Guessing just setting the datatype to the control is enough - 
![image](https://github.com/user-attachments/assets/2fcc82ee-092c-4d6f-af59-ec4d0017d561)

**Reply by mackayn** [reply]
Seems to be, it's a strange warning.
